Sources/Benchmark/BenchmarkResult.swift

```diff
@@ -490,56 +517,127 @@ public struct BenchmarkResult: Codable, Comparable, Equatable {
-            let deviation = ThresholdDeviation(
-                name: name,
-                target: target,
-                metric: metric,
-                percentile: percentile,
-                baseValue: normalize(lhs),
-                comparisonValue: normalize(rhs),
-                difference: Int(Statistics.roundToDecimalplaces(relativeDifference, 1)),
-                differenceThreshold: Int(threshold),
-                relative: true,
-                units: Statistics.Units(timeUnits)
-            )
-            if relativeDifference > threshold {
-                thresholdResults.regressions.append(deviation)
-            } else if relativeDifference < -threshold {
-                thresholdResults.improvements.append(deviation)
+        switch rhs {
+        case .absolute(let rhs):
+            let absoluteDifference = (reverseComparison ? -1 : 1) * (lhs - rhs)
+            let relativeDifference =
+                (reverseComparison ? 1 : -1) * (rhs != 0 ? (100 - (100.0 * Double(lhs) / Double(rhs))) : 0.0)
+
+            if let threshold = thresholdsFromCode.relative[percentile],
+                !(-threshold...threshold).contains(relativeDifference)
+            {
+                let deviation = ThresholdDeviation(
+                    name: name,
+                    target: target,
+                    metric: metric,
+                    percentile: percentile,
+                    baseValue: normalize(lhs),
+                    deviation: .relative(
+                        comparedTo: normalize(rhs),
+                        differencePercentage: Statistics.roundToDecimalPlaces(relativeDifference, 1),
+                        tolerancePercentage: Statistics.roundToDecimalPlaces(threshold, 1)
+                    ),
+                    units: Statistics.Units(timeUnits)
+                )
+                if relativeDifference > threshold {
+                    thresholdResults.regressions.append(deviation)
+                } else if relativeDifference < -threshold {
+                    thresholdResults.improvements.append(deviation)
+                }
             }
-        }
 
-        if let threshold = thresholds.absolute[percentile], !(-threshold...threshold).contains(absoluteDifference) {
-            let deviation = ThresholdDeviation(
-                name: name,
-                target: target,
-                metric: metric,
-                percentile: percentile,
-                baseValue: normalize(lhs),
-                comparisonValue: normalize(rhs),
-                difference: normalize(absoluteDifference),
-                differenceThreshold: normalize(threshold),
-                relative: false,
-                units: Statistics.Units(timeUnits)
-            )
+            if let threshold = thresholdsFromCode.absolute[percentile],
+                !(-threshold...threshold).contains(absoluteDifference)
+            {
+                let deviation = ThresholdDeviation(
```

Comment:
Not quite sure about these SwiftLint errors/warnings. Some of them doesn't play well with swift-format, like this one.

Let me know what I should do.